batch_tasks/resource/hook/batch_tasks.py

```diff
@@ -0,0 +1,385 @@
+            event
+        ]
+
+    def _get_entity_type(self, entity):
+        '''Return translated entity type tht can be used with API.'''
+        entity_type = entity.get('entityType')
+        object_typeid = None
+
+        for schema in self._session.schemas:
+            alias_for = schema.get('alias_for')
+
+            if (
+                alias_for and isinstance(alias_for, dict) and
+                alias_for['id'].lower() == entity_type and
+                object_typeid == alias_for.get(
+                    'classifiers', {}
+                ).get('object_typeid')
+            ):
+
+                return schema['id']
+
+        for schema in self._session.schemas:
+            alias_for = schema.get('alias_for')
+
+            if (
+                alias_for and isinstance(alias_for, basestring) and
+                alias_for.lower() == entity_type
+            ):
+                return schema['id']
+
+        for schema in self._session.schemas:
+            if schema['id'].lower() == entity_type:
+                    return schema['id']
+
+        raise ValueError(
+            'Unable to translate entity type.'
+        )
+
+    def _launch(self, event):
+        args = self._translate_event(
+            self._session, event
+        )
+
+        interface = self._interface(
+            self._session, *args
+        )
+
+        if interface:
+            return interface
+
+        response = self.launch(
+            self._session, *args
+        )
+
+        return self._handle_result(
+            self._session, response, *args
+        )
+
+    def launch(self, session, uid, entities, source, values, event):
+        '''Callback method for the custom action.
+
+        return either a bool (True if successful or False if the action failed)
+        or a dictionary with they keys `message` and `success`, the message
+        should be a string and will be displayed as feedback to the user,
+        success should be a bool, True if successful or False if the action
+        failed.
+
+        *session* is a `ftrack_api.Session` instance
+
+        *uid* is the unique identifier for the event
+
+        *entities* is a list of tuples each containing the entity type and the
+        entity id.
+        If the entity is a hierarchical you will always get the entity
+        type TypedContext, once retrieved through a get operation you
+        will have the "real" entity type ie. example Shot, Sequence
+        or Asset Build.
+
+        *source* dictionary containing information about the source of the
+        event, application id, current user etc.
+
+        *values* is a dictionary containing potential user settings
+        from previous runs.
+
+        *event* the unmodified original event
+
+        '''
+        if 'values' in event['data']:
+            values = event['data']['values']
+            if 'number_of_tasks' in values:
+                items = []
+                TASK_TYPE_ENUMERATOR_OPTIONS = [
```

Comment:
This is a variable, uppercase items typically indicate a static configuration value. This is covered under PEP8.